src/raft/progress.rs

```diff
@@ -41,6 +41,43 @@ impl Default for ProgressState {
     }
 }
 
+/// `ProgressSet` contains several `Progress`es,
+/// which could be `Leader`, `Follower` and `Learner`.
+#[derive(Default, Clone)]
+pub struct ProgressSet {
+    pub voters: FlatMap<u64, Progress>,
```

Comment:
Can they be kept private?

Reply:
test code in `tests/raft/test_raft.rs` also access these fields by `mut`. So we need more `getter` and `mutter`.

Reply:
Because they are always queried by id, I think only one pair of `get` and `get_mut` should be enough.

Reply:
If keep them as private, this logic must be moved into `progress.rs`:
```rust
prs.voters.iter_mut().chain(&mut prs.learners)...
```
Because we can't write code like this:
```rust
prs.mut_voters().chain(prs.mut_learners())
```

And we will also need method
```rust
fn ProgressSet::for_each<F: FnMut(ProgressSet)>(u64, f: F) {
    // f is a closure which will modify both captured things and `ProgressSet` itself.
}
```

Is it a little wordy?